### PR TITLE
fixes scrubbers being unable to scrub

### LINKED ISSUE #32815: Air Scrubbers Don't Scrub
The new Atmospherics update came with a few changes, including how the scrubbers work. The code change which allows it to filter out rare gasses such as BZ, Tritium, and Pluoxium individually broke them. The siphoning mode works just fine for the SM loop, which was probably tested because if it did not do its job, the SM would blow up. However, the scrubbing mode is broken. When a scrubber is set to scrubbing, it does not filter out any gasses.
image
There are probably more problems with the new scrubbers, but this is the only issue I could prove without a shadow of a doubt with testing.

### PR DESCRIPTION
**Pull request by duncathan**
Fixes #32815

### CODE REVIEW

File: code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm [hunk 1]
```diff
@@ -163,14 +163,7 @@
 		return FALSE
 
 	if(scrubbing & SCRUBBING)
-		var/should_we_scrub = FALSE
-		for(var/id in env_gases)
-			if(id == /datum/gas/nitrogen || id == /datum/gas/oxygen)
-				continue
-			if(env_gases[id][MOLES] && filter_types[id])
```
Comment (duncathan) [inline code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm]: this is the line that broke it; `filter_types` isn't actually an associative list so `filter_types[id]` was null even if `id in filter_types` were true

**Comment by davethwave** [comment]
Thanks for doing this.

**Comment by duncathan** [comment]
I broke it in the first place lol

**Comment by davethwave** [comment]
But still, at least you took time to fix it.

**Pull request merged**